homeassistant/components/zabbix/__init__.py

```diff
@@ -46,13 +90,208 @@ def setup(hass, config):
 
     hass.data[DOMAIN] = zapi
+
+    def event_to_metrics(event, float_keys, string_keys):
+        """Add an event to the outgoing Zabbix list."""
+        state = event.data.get("new_state")
+        if (
+            state is None
+            or state.state in (STATE_UNKNOWN, "", STATE_UNAVAILABLE)
+            or state.entity_id in blacklist_e
+            or state.domain in blacklist_d
+        ):
+            return
+
+        entity_id = state.entity_id
+        floats = {}
+        strings = {}
+        try:
+            if (
+                (whitelist_e or whitelist_d)
+                and entity_id not in whitelist_e
+                and state.domain not in whitelist_d
+            ):
+                return
+
+            _state_as_value = float(state.state)
+            floats[entity_id] = _state_as_value
+        except ValueError:
+            try:
+                _state_as_value = float(state_helper.state_as_number(state))
+                floats[entity_id] = _state_as_value
+            except ValueError:
+                strings[entity_id] = state.state
+
+        for key, value in state.attributes.items():
+            # For each value we try to cast it as float
+            # But if we can not do it we store the value
+            # as string
+            attribute_id = f"{entity_id}/{key}"
+            try:
+                float_value = float(value)
+                if math.isfinite(float_value):
+                    floats[attribute_id] = float_value
+                else:
+                    strings[attribute_id] = str(value)
+            except (ValueError, TypeError):
+                strings[attribute_id] = str(value)
+
+        metrics = []
+        float_keys_count = len(float_keys)
+        float_keys.update(floats.keys())
+        if len(float_keys) != float_keys_count:
+            floats_discovery = []
+            for float_key in float_keys:
+                floats_discovery.append({"{#KEY}": float_key})
+            metric = ZabbixMetric(
+                publish_states_host,
+                "homeassistant.floats_discovery",
+                json.dumps(floats_discovery),
+            )
+            metrics.append(metric)
+        for key, value in floats.items():
+            metric = ZabbixMetric(
+                publish_states_host, f"homeassistant.float[{key}]", value
+            )
+            metrics.append(metric)
+
+        string_keys.update(strings.keys())
+        return metrics
+
+    def shutdown(event):
+        """Shut down the thread."""
+        instance.queue.put(None)
+        instance.join()
+
+    if publish_states_host:
+        zabbix_sender = ZabbixSender(zabbix_server=conf[CONF_HOST])
+        _LOGGER.info("Initialized Zabbix sender")
+        instance = ZabbixThread(hass, zabbix_sender, event_to_metrics, max_tries)
+        instance.start()
+        _LOGGER.info("Started ZabbixThread")
```

Comment:
```suggestion
        _LOGGER.debug("Started ZabbixThread")
```

Reply:
Are these logs really useful? Maybe one of them is enough?

Reply:
I removed both logs.